sklearn/impute/tests/test_impute.py

```diff
@@ -1383,3 +1383,30 @@ def test_imputation_order(order, idx_order):
                                random_state=0).fit(X)
         idx = [x.feat_idx for x in trs.imputation_sequence_]
         assert idx == idx_order
+
+
+def test_simple_imputation_inverse_transform():
+    # Test inverse_transform feature for np.nan
+    X_1 = np.array([
+        [9, np.nan, 3, -1],
+        [4, -1, 5, -1],
+        [6, 7, np.nan, -1],
+        [8, 9, 0, np.nan]
+    ])
+    X_2 = np.array([
```

Comment:
I would try 2 more arrays: one where all columns have missing data and one where there is missing data one column every two

Reply:
Thanks for spotting this!
I overlooked a major scenario where features get dropped for being completely empty in the `transform` step. The process of regenerating original data is not very elegant when columns are dropped resulting in a different shape. I think changes suggested by PR #16695 would make things easier here. 
The logic I wrote right now is iterative. It works only when both the fitting data and transform data are compatible i.e., both have values missing in same features. Let me know if there's a better way to handle this limitation.

Reply:
what I mean was more 

```
    X_3 = np.array([
        [1, missing_value, 5, 9],
        [missing_value, 4, missing_value, missing_value],
        [2, missing_value, 7, missing_value],
        [missing_value, 3, missing_value, 8]
    ])

    X_4 = np.array([
        [1, 1, 1, 3],
        [missing_value, 2, missing_value, 1],
        [2, 3, 3, 4],
        [missing_value, 4, missing_value, 2]
    ])
```

Reply:
Thank you. I update the test cases